### PR TITLE
Fix #336 (partially) Add logic to exclude unlawful Strings 

### LINKED ISSUE #336: monocle.std.string.stringToBoolean doesn't obey prism laws
```
scala> import scalaz.Scalaz._
import scalaz.Scalaz._

scala> monocle.std.string.stringToBoolean.modify(identity)("TRUE") == "TRUE"
res0: Boolean = false

scala> monocle.std.string.stringToBoolean.modify(identity)("TRUE") === "TRUE"
res1: Boolean = false
```

### PR DESCRIPTION
**Pull request by pvillega**
For String-to-Long Prism and derivatives (String-to-Int, String-to-Byte)

### CODE REVIEW

File: core/shared/src/main/scala/monocle/std/String.scala [hunk 1]
```diff
@@ -81,13 +81,19 @@ trait StringOptics {
...
+  private def parseLong(s: String): Option[Long] = {
+    // we reject case where String starts with +, otherwise it will be an invalid Prism according 2nd Prism law
+    // we reject case where String starts with 0 and has multiple digits, otherwise it will be an invalid Prism according 2nd Prism law
+    def inputBreaksPrismLaws(input: String): Boolean =
+      s.isEmpty || s.startsWith("+") || (s.startsWith("0") && s.length > 1)
+
+    if (inputBreaksPrismLaws(s)) None
     else s.toList match {
       case '-' :: xs => parseLongUnsigned(xs).map(-_)
-      case xs        => parseLongUnsigned(xs)
```
Comment (julien-truffaut) [inline core/shared/src/main/scala/monocle/std/String.scala]: would you mind to keep the vertical alignment

**Comment by julien-truffaut** [comment]
just a minor comment, otherwise :+1:

**Comment by julien-truffaut** [comment]
Awesome thanks :)

**Comment by pvillega** [comment]
A pleasure :)

**Pull request merged**